lib/rspec/core/option_parser.rb

```diff
@@ -140,8 +142,23 @@ def parser(options)
           options[:full_backtrace] = true
         end
 
-        parser.on('-c', '--[no-]color', '--[no-]colour', 'Enable color in the output.') do |o|
-          options[:color] = o
+        parser.on('-c', '--color', '--colour', 'Deprecated: enable color in the output.') do |_o|
+          options[:color] = true
+          options[:color_mode] = :automatic
+        end
```

Comment:
Actually, I see now that you did remove it from the output by adding `-c` to the list of ~~`invalid_options`~~ `hidden_options`. Somehow I missed that.

Still, the fact that the code is here makes it easy to think that `--color` _will_ be listed in the output (as I initially thought it would), which makes this approach a bit confusing.  Instead, can you handle it similar to how [`--tty` is processed](https://github.com/rspec/rspec-core/blob/c3c90c278fda997e0eaf3bdac08d162175dc060b/lib/rspec/core/option_parser.rb#L21)?  It's much easier to understand that it is a "hidden" option that does not show up in the `--help` output.